Adding persistance

jhb15 opened the pull request:
Got all code in place for getting/writing/updating/deleting Token to UserId mappings. Also some code in place for getting/writing/updating/deleting AppId and AppSecret. I have added tests to test this code which should pass locally if a DB connection is setup but I had to modify them so they would not run(skip) if it can't connect to a database.

JackThomson2 commented on pom.xml:
```diff
@@ -8,5 +8,71 @@
+            <artifactId>postgresql</artifactId>
+            <version>42.2.5.jre7</version>
+        </dependency>
+        <!--<dependency>
```
I presume we don't need this dependency?

jhb15 replied:
Oh yeah I only needed it for the local DB i had setup.

JackThomson2 commented:
It also looks like Travis is skipping all the tests.

dkmonaghan commented on fitbit-ingest-service.iml:
```diff
@@ -12,15 +12,61 @@
+    <orderEntry type="library" name="Maven: org.dom4j:dom4j:2.1.1" level="project" />
+    <orderEntry type="library" name="Maven: org.hibernate.common:hibernate-commons-annotations:5.0.4.Final" level="project" />
+    <orderEntry type="library" name="Maven: org.hibernate:hibernate-entitymanager:5.3.7.Final" level="project" />
+    <orderEntry type="library" name="Maven: org.postgresql:postgresql:42.2.5.jre7" level="project" />
```
Update your branch without your testing dependencies - pull request will automatically update itself once you update the adding_persistence branch.

jhb15 replied:
removed the postgresql testing dependency but the hibernate stuff isn't for testing its required.

dkmonaghan commented on pom.xml:
```diff
@@ -8,5 +8,71 @@
+            <version>5.3.7.Final</version>
+        </dependency>
+        <dependency>
+            <groupId>org.postgresql</groupId>
```
As above.

jhb15 replied:
done

dkmonaghan commented on src/main/java/persistance/ClientCredentials.java:
```diff
@@ -0,0 +1,62 @@
+import javax.persistence.*;
+
+@Entity(name = "client_credentials")
+public class ClientCredentials {
```
All of this class is missing Javadoc comments - please could you add them

jhb15 replied:
done

dkmonaghan commented on src/main/java/persistance/StorageManager.java:
```diff
@@ -0,0 +1,218 @@
+     * @param uId user id of the current user
+     * @return 0 for success, -1 failure (record doesn't exist)
+     */
+    public int removeTokenMap(String uId) {
```
Why does this return an integer to determine failure / success? Can we not use a true/false bool here?

jhb15 replied:
done

dkmonaghan commented on src/main/java/persistance/StorageManager.java:
```diff
@@ -0,0 +1,218 @@
+        return tokeMap;
+    }
+
+    private void createTokenMap(String uID, String accessToken, String refreshToken) {
```
Missing Javadoc comments

jhb15 replied:
done

dkmonaghan commented on src/main/java/persistance/StorageManager.java:
```diff
@@ -0,0 +1,218 @@
+        }
+    }
+    
+    private void updateTokenMap(TokenMap tm, String accessToken, String refreshToken) {
```
Missing Javadoc comments

dkmonaghan commented on src/main/java/persistance/StorageManager.java:
```diff
@@ -0,0 +1,218 @@
+        em.getTransaction().commit();
+    }
+
+    private TokenMap getTokenMap(String uId) {
```
Missing Javadoc comments

dkmonaghan commented on src/main/java/persistance/StorageManager.java:
```diff
@@ -0,0 +1,218 @@
+     * This funtion removes Application Credentials
+     * @return
+     */
+    public int removeAppCreds() {
```
Again, could this be a boolean? Javadoc comment also needs finishing off.

dkmonaghan commented on src/main/java/persistance/StorageManager.java:
```diff
@@ -0,0 +1,218 @@
+        return clientCredentials;
+    }
+
+    private void  updateCredRecord(ClientCredentials cred, String id, String secret) {
```
Missing Javadoc comments

dkmonaghan commented on src/main/java/persistance/StorageManager.java:
```diff
@@ -0,0 +1,218 @@
+        em.getTransaction().commit();
+    }
+
+    private void createCredRecord(String id, String secret) {
```
Missing Javadoc comments

jhb15 replied:
done

dkmonaghan commented on src/main/java/persistance/StorageManager.java:
```diff
@@ -0,0 +1,218 @@
+        em.getTransaction().commit();
+    }
+
+    private ClientCredentials getClientCredentials() {
```
Missing Javadoc comments

dkmonaghan commented on src/main/java/persistance/TokenMap.java:
```diff
@@ -0,0 +1,63 @@
+import java.io.Serializable;
+
+@Entity(name = "token_map")
+public class TokenMap implements Serializable {
```
This class is missing Javadoc comments throughout.

dkmonaghan commented on src/main/java/persistance/readme.txt:
```diff
@@ -0,0 +1,17 @@
+How to Connect Local DB:
```
This information is probably going to get overlooked if this readme.txt is in this directory. I'd recommend you make a 'docs' folder in the root of your repository and put this file there. 

(also, give it a .md file extension so GitHub renders it nicely)

dkmonaghan commented on src/main/resources/META-INF/persistence.xml:
```diff
@@ -0,0 +1,21 @@
+        <class>persistance.TokenMap</class>
+        <class>persistance.ClientCredentials</class>
+        <properties>
+            <property name="hibernate.connection.url" value="jdbc:postgresql://localhost:5432/fitbit_digest_db"/>
```
Hardcoded PostgreSQL connection information - probably needs removing too?

dkmonaghan commented:
> It also looks like Travis is skipping all the tests. Not sure how we want to tackle that.

@DavidFair any ideas?

DavidFair commented:
The tests were skipped:
> Tests run: 6, Failures: 0, Errors: 0, Skipped: 6

I suspect it is related to these warnings, but I would need to put a breakpoint on a test to see if they are even entered:
```
Oct 30, 2018 10:28:50 PM org.hibernate.engine.jdbc.connections.internal.DriverManagerConnectionProviderImpl configure
WARN: HHH10001002: Using Hibernate built-in connection pool (not for production use!)
Oct 30, 2018 10:28:50 PM org.hibernate.engine.jdbc.connections.internal.DriverManagerConnectionProviderImpl buildCreator
INFO: HHH10001005: using driver [org.postgresql.Driver] at URL [jdbc:postgresql://localhost:5432/fitbit_digest_db]
Oct 30, 2018 10:28:50 PM org.hibernate.engine.jdbc.connections.internal.DriverManagerConnectionProviderImpl buildCreator
INFO: HHH10001001: Connection properties: {user=james, password=****}
Oct 30, 2018 10:28:50 PM org.hibernate.engine.jdbc.connections.internal.DriverManagerConnectionProviderImpl buildCreator
INFO: HHH10001003: Autocommit mode: false
Oct 30, 2018 10:28:50 PM org.hibernate.engine.jdbc.connections.internal.DriverManagerConnectionProviderImpl$PooledConnections <init>
INFO: HHH000115: Hibernate connection pool size: 20 (min=1)
Oct 30, 2018 10:28:50 PM org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator initiateService
WARN: HHH000342: Could not obtain connection to query metadata : null
```

jhb15 commented:
I have now added the JavaDoc comments and also removed the -1, 0 return messages. I also removed that hard-coded DB properties and left them empt for now as that didn't cause a build failure. I also removed the PostgreSQL driver dependency I was using for testing.
